src/Wallclock.h

```diff
@@ -0,0 +1,137 @@
+/*++
+
+Module name:
+
+    Wallclock
+    
+Abstract:
+
+    Provides methods to update and retrieve timestamps, calculate elapsed
+    time, and convert time into various units (such as seconds or hours).
+    The class also includes a constant to store a timestamp during the game's
+    initialization, allowing for easy calculations of the time elapsed since
+    the game start.
+    
+Notes:
+
+    A few tools are provided here to allow you to do virtually anything
+    regarding time calculations, such as getting the time since the game
+    started, calculating the time elapsed between two timestamps, and
+    converting time into different units.
+
+    Wallclock was intended to provide more accurate timestamps for time-
+    critical functions, but the functions it provides can be used to do
+    just about anything RageTimer
+    can do.
+
+    A static constant (gamestart) that stores the time when the game started
+    is provided so you can make manual calculations based on the game's start
+    time anywhere Wallclock is included.
+    
+Usage example:
```

Comment:
I know this usage example section needs work, I will make it less awkward if this is to be merged.